Adding exception text to exit message for invalid clientsecrets

pferate opened the pull request:
This is to address a problem found in google/google-api-python-client#228.

This will give users the text from the InvalidClientSecretsError exception, in addition to the generic message that it currently does. The format of the message can be modified, if needed.

theacodes commented on oauth2client/client.py:
```diff
@@ -2210,8 +2210,10 @@ def flow_from_clientsecrets(filename, scope, redirect_uri=None,
         if message:
+            if str(e):
+                message = 'Error: %s\n================\n%s' % (str(e), message)
             sys.exit(message)
```
Holy cow, I'm really appalled we have a `sys.exit` hiding in here. D:

theacodes commented on oauth2client/client.py:
```diff
@@ -2210,8 +2210,10 @@ def flow_from_clientsecrets(filename, scope, redirect_uri=None,
-    except clientsecrets.InvalidClientSecretsError:
+    except clientsecrets.InvalidClientSecretsError as e:
         if message:
+            if str(e):
```
`if e.args`?

pferate replied:
better.  I'll update it.

theacodes replied:
I'm unsure about py2.6 / py3 compatibility here so tread carefully.

pferate replied:
Just tested this, it works in at least 2.6, 2.7, 3.2, 3.3, 3.4, and 3.5.

theacodes replied:
Thanks.

theacodes commented on oauth2client/client.py:
```diff
@@ -2210,8 +2210,10 @@ def flow_from_clientsecrets(filename, scope, redirect_uri=None,
+    except clientsecrets.InvalidClientSecretsError as e:
         if message:
+            if str(e):
+                message = 'Error: %s\n================\n%s' % (str(e), message)
```
I'd prefer just `'The client secrets were invalid: \n {}'.format(e)`

pferate replied:
Sounds better.  What about if message is passed as a parameter?

theacodes replied:
Works just as well.

jonparrott commented:
Thanks for this contribution. Do you mind adding or augmenting the existing test for this?

nathanielmanistaatgoogle commented on oauth2client/client.py:
```diff
@@ -2210,8 +2210,10 @@ def flow_from_clientsecrets(filename, scope, redirect_uri=None,
 
-    except clientsecrets.InvalidClientSecretsError:
+    except clientsecrets.InvalidClientSecretsError as e:
         if message:
```
While we're here: `if message is not None:`.

pferate replied:
will do.

pferate committed:
Add exception text to exit message

If `clientsecrets.InvalidClientSecretsError` is raised with additional
text and a predefined failure message is given, include exception text
in the exit message.

pferate commented:
@jonparrott @nathanielmanistaatgoogle, I updated this PR again and added a test for this (slightly modified a previous test).  Travis passed on my branch, but still waiting to launch for this PR.

I had to use positional argument specifiers in the `.format()` string to have it work with 2.6.

jonparrott commented:
Great, this looks good to me. @nathanielmanistaatgoogle to do the final review.

nathanielmanistaatgoogle commented:
Change content looks good; [please amend your commit log message to meet the guidelines](http://chris.beams.io/posts/git-commit/#seven-rules).

pferate commented:
Commit message updated.

jonparrott commented:
Waiting for travis.

jonparrott commented:
Thanks for your contributions, @pferate!
